### PR TITLE
[lwp][vdso] Add VDSO functionality under the aarch64 architecture

### PR DESCRIPTION
**Pull request by rcitachi**
## 拉取/合并请求描述：(PR description)

[

#### 请提供验证的bsp和config (provide the config and bsp) 

- BSP:

- .config:

- action:

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

**Comment by polarvid** [comment]
Relates https://github.com/RT-Thread/rt-thread/pull/9158

File: components/lwp/arch/aarch64/common/vdso.c [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    }
+    vdso_info[VDSO_ABI_AA64].vdso_pages = (
+        vdso_info[VDSO_ABI_AA64].vdso_code_end -
+        vdso_info[VDSO_ABI_AA64].vdso_code_start) >>
+        ARCH_PAGE_SHIFT;
+
+    __initdata();
+    return RT_EOK;
+}
+INIT_COMPONENT_EXPORT(validate_vdso_elf);
```
Comment (polarvid) [inline components/lwp/arch/aarch64/common/vdso.c:110]: 初始化似乎不需要全局可见吧？

File: components/lwp/vdso/vdso_config.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define __VVAR_PAGES    2
+#define VDSO_PAGE_SHIFT     12
+#define VDSO_PAGE_SIZE      (1 << VDSO_PAGE_SHIFT)
+
+#define BIT(nr)         ((1) << (nr))
```
Comment (polarvid) [inline components/lwp/vdso/vdso_config.h:22]: 这个名字换一个吧，太容易冲突了

**Review by polarvid — CHANGES_REQUESTED** [review]

File: components/lwp/arch/aarch64/common/vdso.c (hunk 1 above)
Comment (rcitachi) [reply components/lwp/arch/aarch64/common/vdso.c:110]: 这个确实不需要。没有找到一个合适的地方初始化，就直接用自动初始化了

Comment (polarvid) [reply components/lwp/arch/aarch64/common/vdso.c:110]: 所以，为什么不加一个 static 限定符呢

Comment (rcitachi) [reply components/lwp/arch/aarch64/common/vdso.c]: 👌

**Review by polarvid — APPROVED** [review]

**Review by BernardXiong — APPROVED** [review]

**Review by BernardXiong — APPROVED** [review]

**Pull request merged**